### PR TITLE
ci: rename commitlint.config.js to commitlint.config.mjs

### PR DESCRIPTION
**Pull request by jan--f**
This fixes commitlint failures caused by
https://github.com/wagoid/commitlint-github-action/pull/763.

### CODE REVIEW

File: commitlint.config.mjs
Comment (danielmellado) [inline commitlint.config.mjs:1]: We would need to change module.exports to `export default Configuration;` in here to comply with the ES module syntax.

Comment (jan--f) [reply commitlint.config.mjs:1]: Like so?

Comment (danielmellado) [reply commitlint.config.mjs:1]: yeah, I think that should make it.

File: commitlint.config.mjs [hunk 1]
```diff
@@ -9,4 +9,4 @@ const Configuration = {
   ignores: [(msg) => /Signed-off-by: dependabot\[bot]/m.test(msg)],
 };
 
-module.exports = Configuration;
+export default Configuration
```
Comment (danielmellado) [inline commitlint.config.mjs:12]:
```suggestion
export default Configuration;
```

Comment (danielmellado) [reply commitlint.config.mjs:12]: let's keep the end trailing even if it does seem that it works without it.

**Pull request merged**